packages/jamulus/PKGBUILD

```diff
@@ -0,0 +1,50 @@
+# Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
+# Contributor: Jose Riha <jose1711 gmail com>
+# Contributor: Julien Taverna <jujudusud gmail com>
+# Contributor: Florian Hülsmann <fh@cbix.de>
+
+pkgbase=jamulus
+pkgname=(jamulus jamulus-server)
+pkgver=3.8.2
+pkgrel=2
+pkgdesc="Internet jam session software"
+arch=(x86_64 aarch64)
+url='https://jamulus.io/'
+license=(GPL2)
+depends=(gcc-libs qt5-base)
+provides=(jamulus-server)
+conflicts=(jamulus-git)
+makedepends=(jack libjack.so)
+source=("${pkgbase}-${pkgver}.tar.gz::https://github.com/jamulussoftware/${pkgbase}/archive/r${pkgver//./_}.tar.gz")
+sha512sums=('8630cbcbfbdc2051ecbbd5158211efce2f2e47ae18413cb7e549bbe8909110ff13ab8ac2dbacd8c41a88b38d9988ea7c3955a6fec25a1eb18170ab6d792063c7')
+_pkgsrc=${pkgbase}-r${pkgver//./_}
+
+build() {
+  cd "$srcdir"/${_pkgsrc}
+  qmake "CONFIG+=nosound headless" TARGET=jamulus-server
+  make clean
+  make
+  qmake "CONFIG+=noupcasename"
+  make
+}
+
+package_jamulus() {
+  cd "$srcdir"/${_pkgsrc}
+  pkgdesc+=" (client and server)"
+  depends+=(libjack.so)
+  provides+=(jamulus)
+  install -vDm755 jamulus "$pkgdir"/usr/bin/jamulus
+  install -vDm644 ChangeLog README.md -t "$pkgdir"/usr/share/doc/${pkgbase}
+  install -vDm644 COPYING -t "$pkgdir"/usr/share/licenses/${pkgbase}
+  install -vDm644 distributions/jamulus{,-server}.desktop -t "$pkgdir"/usr/share/applications
+  install -vDm644 distributions/jamulus{,-server}.svg -t "$pkgdir"/usr/share/pixmaps
+}
+
+package_jamulus-server() {
```

Comment:
The client and server come in one single binary, so in `jamulus` the binary provides client + server (graphical + headless) and in `jamulus-server` it provides only a headless server.

Reply:
I can understand the motivation to put this in a separate package, so one can only install `jamulus-server` on a minimal "jamulus box", but still, why not allow both to be installed together?

Reply:
My main concern is that users will understand what to install when they want one or the other. In the end if you do `pacman -Ss jamulus` and you'll see `jamulus` + `jamulus-server` it's mainly up to the `$pkgdesc` to communicate what to install when.
```
$ pacman -Ss jamulus
proaudio/jamulus 3.8.2-2
    Internet jam session software (client and server)
proaudio/jamulus-server 3.8.2-2
    Internet jam session software (headless server)
```

Technically it just doesn't make sense to install both packages but I'll think about this again.

Reply:
I may not make sense to run both server versions at the same time, but IMHO you may still want to run the UI version at one point in time and the headless version at another on the same machine. It would be tedious to have to re-install packages each time.

Reply:
again: the headless server binary provides _no benefit_ over the _normal_ binary if they're on the same machine. It's really just linking less stuff :D

In your scenario everything will just keep working:
1. user installs `jamulus-server`
2. user starts the server by running `jamulus -s ...`
3. user later decides they want to run the client on the same machine, so they install `jamulus`
4. the server is still running in memory, but may even be restarted with _the same_ command without any side effects

The other way around is even simpler:
1. user installs `jamulus` and launches the client
2. user is not aware that the server is included already and installs `jamulus-server`
3. pacman gracefully reminds you that `jamulus-server` is already satisfied
4. user can already start the headless server using `jamulus -s ...`

Reply:
I see, I overlooked the "headless" in "the binary provides client + server (graphical + headless)". I wonder then, whether the headless package should not be named `jamulus-headless` or `jamulus-headless-server`? Otherwise users might think they can install `jamulus-server` to run a server with the GUI.

Reply:
Cool, I'll talk to @jujudusud who is currently co-maintaining the `jamulus-headless` AUR so we may find a solution that won't break too many people's setup, should they eventually switch to our binary repo.

Reply:
Jamulus binary is actually written with an upcase 'J'.
There are four different binaries available from config :
- Client and server with Qt GUI,
- Client and server without Qt GUI (headless),
- Server only with Qt GUI but without needing JACK and the all the sound stuff (nosound),
- Server only without Qt GUI, JACK and all the sound stuff (nosound headless).